### ISSUE TITLE
Some Questions About PPFT Stage

### ISSUE THREAD

**Issue by Xiao-Fangff** [Georgefwt/AquaLoRA#5]
Hi! I've read the part of ppft stage in paper and code. I wonder why the watermark remains preserved during the denoising phase if we ensure the predicted noise $\bar{\epsilon}_{pred}$ and $$\epsilon{pred}$$ are similar. In other words, how does the training methodology ensure the preservation of the watermark, despite the absence of any watermark-related loss during the PPFT phase?

![image](https://github.com/user-attachments/assets/4d589e2c-3833-40bd-8d17-b301e098ec49)

**Comment by Georgefwt** [comment]
This is an interesting problem. It can be observed that the training input for the diffusion model (including LoRA) contains watermarked samples, while the input for the frozen diffusion model consists of clean samples. It is through this distinction that the watermark information is introduced. A detailed proof can be found in Appendix E of our [paper](https://arxiv.org/pdf/2405.11135v1).

**Comment by qsong2001** [comment]
Maybe it is like the watermarkDM [1], using watermarked data to fine-tune the network.  
[1] https://arxiv.org/pdf/2303.10137v1

**Comment by Georgefwt** [comment]
This is a simple way to understand this loss, but I should point out that methods like watermarkDM, which directly use watermarked data for fine-tuning, can easily cause the model's distribution to shift toward the training data. The method we propose, however, won't be significantly affected by the training set.

**Comment by qsong2001** [comment]
This work inspires me a lot 😄
